src/hotspot/share/logging/logAsyncWriter.cpp

```diff
@@ -174,7 +203,17 @@ void AsyncLogWriter::run() {
       });
       _data_available = false;
     }
+
     write(snapshot);
+
+    if (_stalled_message != nullptr) {
```

Comment:
Just an observation here that reading this without the lock is only valid because only this thread ever nulls out the stalled message - hence a non-null stalled message must have come from another thread, and we then take the lock and so ensure we see the stalled message correctly.

Reply:
Correct.